dipy/denoise/denspeed.pyx

```diff
@@ -108,9 +103,10 @@ def _nlmeans_3d(double[:, :, ::1] arr, double[:, :, ::1] mask,
 
     # move the block
     with nogil, parallel():
-        for i in prange(B + P, I - B - P, schedule="dynamic"):
```

Comment:
I believe that you can retain this change (introduced here: https://github.com/nipy/dipy/commit/4ee6d49a03474b512a6c1c912a9e38c717f3d1f7) without issue. As I understand it, this only (substantially!) speeds up performance without change in the results. Worth making sure, though.